File: Sunrin-ICPC-2021/Final-B/FB.java
import java.util.Scanner;

public class Main{
    public static long pow(long a, long b, long mod){
        long res = 1;
        while(b != 0){
            if(b % 2 == 1) res = res * a % mod;
            a = a * a % mod;
            b /= 2;
        }
        return res;
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        long n = sc.nextLong();
        long k = sc.nextLong();
        long x = sc.nextLong();
        long mod = 998244353;

        long sum = (x * (x + 1) / 2) % mod;
        long res = pow(x, n+k-2, mod) * sum % mod * sum % mod * k % mod;
        for(int i=0; i<n-k+1; i++) System.out.printf("%d ", res);
    }
}

File: Sunrin-ICPC-2021/Final-B/FB.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
constexpr ll MOD = 998244353;
// X^{N+K-2} * sum(1..X)^2 * K

//get a^b mod MOD in O(log b)
ll Pow(ll a, ll b){
    ll res = 1;
    for(; b; b >>= 1, a = a * a % MOD) if(b & 1) res = res * a % MOD;
    return res;
}

int main(){
    ios_base::sync_with_stdio(false); cin.tie(nullptr);
    ll N, K, X; cin >> N >> K >> X;
    ll S = X * (X + 1) / 2 % MOD;
    ll R = Pow(X, N+K-2) * Pow(S, 2) % MOD * K % MOD;
    for(int i=1; i<=N-K+1; i++) cout << R << " ";
}